File: src/main/java/com/kopyl/commit/JiraClient.java
package com.kopyl.commit;

import com.atlassian.jira.rest.client.api.JiraRestClientFactory;
import com.atlassian.jira.rest.client.api.JiraRestClient;
import com.atlassian.jira.rest.client.api.domain.Issue;
import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory;

import java.net.URI;

public class JiraClient {
    private final String token;
    private final String jiraUrl;
    private final JiraRestClient restClient;

    public JiraClient(String token, String jiraUrl) {
        this.token = token;
        this.jiraUrl = jiraUrl;
        this.restClient = getJiraRestClient();
    }

    private JiraRestClient getJiraRestClient() {
        BearerHttpAuthenticationHandler handler = new BearerHttpAuthenticationHandler(this.token);
        JiraRestClientFactory factory = new AsynchronousJiraRestClientFactory();
        return factory.create(getJiraUri(), handler);
    }

    private URI getJiraUri() {
        return URI.create(this.jiraUrl);
    }

    public Issue getIssue(String issueKey) {
        return restClient.getIssueClient()
            .getIssue(issueKey)
            .claim();
    }
}

File: src/main/kotlin/com/kopyl/commit/JiraClient.kt
package com.kopyl.commit

import com.atlassian.jira.rest.client.api.JiraRestClient
import com.atlassian.jira.rest.client.api.JiraRestClientFactory
import com.atlassian.jira.rest.client.api.domain.Issue
import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory
import java.net.URI

class JiraClient(private val token: String, private val jiraUrl: String) {
    private val restClient: JiraRestClient

    init {
        restClient = jiraRestClient
    }

    private val jiraRestClient: JiraRestClient
        private get() {
            val handler = BearerHttpAuthenticationHandler(token)
            val factory: JiraRestClientFactory = AsynchronousJiraRestClientFactory()
            return factory.create(jiraUri, handler)
        }
    private val jiraUri: URI
        private get() = URI.create(jiraUrl)

    fun getIssue(issueKey: String?): Issue {
        return restClient.issueClient
                .getIssue(issueKey)
                .claim()
    }
}
